tools/TagMatcher/tag-matcher.py

```diff
@@ -74,9 +74,14 @@ def has_mismatch(match_list):
             with io.open(file_path, 'r') as f:
                 # For each file, generate the match dictionary.
                 matches[file_path] = populate_match_list()
-                for x in f:
-                    # Then for each line in the file, conduct the tuple open/close matching.
-                    get_tag_matches(matches[file_path], x)
+                try:
+                    for x in f:
+                        # Then for each line in the file, conduct the tuple open/close matching.
+                        get_tag_matches(matches[file_path], x)
+                except UnicodeDecodeError:
+                    print("There is a non UTF-8 format character in {}".format(f))

```

Comment:
More information would be useful.

```suggestion
                    print("There is a non UTF-8 format character in file {}. Line:\n\r{}".format(f, x))
```

Reply:
X is not withing same scope as print statement